packages/endo/src/import.js

```diff
@@ -0,0 +1,58 @@
+/* global StaticModuleRecord */
+/* eslint no-shadow: 0 */
+
+import { compartmentMapForNodeModules } from "./compartmap.js";
+import { search } from "./search.js";
+import { assemble } from "./assemble.js";
+
+const decoder = new TextDecoder();
+
+const resolve = (rel, abs) => new URL(rel, abs).toString();
+
+const makeImportHookMaker = (read, rootLocation) => packageLocation => {
+  packageLocation = resolve(packageLocation, rootLocation);
+  return async moduleSpecifier => {
+    const moduleLocation = resolve(moduleSpecifier, packageLocation);
+    const moduleBytes = await read(moduleLocation);
+    const moduleSource = decoder.decode(moduleBytes);
+    return new StaticModuleRecord(moduleSource, moduleLocation);
+  };
+};
+
+export const loadPath = async (read, modulePath) => {
+  const {
+    packageLocation,
+    packageDescriptorText,
+    moduleSpecifier
+  } = await search(read, modulePath);
+
+  const packageDescriptor = JSON.parse(packageDescriptorText);
+  const compartmentMap = await compartmentMapForNodeModules(
+    read,
+    packageLocation,
+    [],
+    packageDescriptor
+  );
+
+  const { compartments, main } = compartmentMap;
+
+  const makeImportHook = makeImportHookMaker(read, packageLocation);
+
+  const execute = async (endowments, modules) => {
+    const compartment = assemble({
+      name: main,
+      compartments,
+      makeImportHook,
+      endowments,
+      modules
+    });
+    return compartment.import(moduleSpecifier);
```

Comment:
If I didn't suspect that `compartment.import` returns a Promise, I might be confused as to why `execute` is marked `async`. But I think marking it async is a fine way to remind callers what to expect.